api/batch/controller.go

```diff
@@ -158,12 +158,13 @@ func (c *controller) Submit(ctx context.Context, predictionJob *models.Predictio
 		return fmt.Errorf("failed creating spark driver authorization in namespace %s: %w", namespace, err)
 	}
 
-	secret, err := c.mlpAPIClient.GetSecretByName(ctx, predictionJob.Config.ServiceAccountName, int32(predictionJob.ProjectID))
+	// Get MLP secrets for Google service account and user-specified MLP secrets
+	secretMap, err := c.getMLPSecrets(ctx, predictionJob, namespace)
 	if err != nil {
-		return fmt.Errorf("service account %s is not found within %s project: %w", predictionJob.Config.ServiceAccountName, namespace, err)
+		return fmt.Errorf("error retrieving secrets: %w", err)
 	}
 
-	_, err = c.manifestManager.CreateSecret(ctx, predictionJob.Name, namespace, secret.Data)
+	_, err = c.manifestManager.CreateSecret(ctx, predictionJob.Name, namespace, secretMap)
```

Comment:
Q: do different batch prediction jobs have unique names? In case a new batch prediction job is submitted before the previous one completes, the secret names should be unique right?

Reply:
Yep, they do, the `predictionJob.Name` comes from `jobName` here https://github.com/caraml-dev/merlin/blob/967788eb98a0120dd20bec692ab04bcea8f08fc9/api/service/prediction_job_service.go#L145 which itself is generated using a (supposedly) unique id as a suffix for each job run `strconv.FormatInt(p.clock.Now().UnixNano(), 10)[:13]`.

Reply:
ah ok cool thanks!